python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -428,15 +432,41 @@ def evaluate_asset(
                 dynamic_partitions_store=self.instance_queryer,
             ),
             to_materialize,
+            to_skip,
             to_discard,
         )
 
+    @functools.cached_property
+    def skipped_asset_graph_subset(self) -> AssetGraphSubset:
+        subset = self.cursor.skipped_asset_graph_subset or AssetGraphSubset(self.asset_graph)
+        # factor out any asset partitions which have been materialized since last tick
```

Comment:
I don't think I would expect a function with this name to omit these. Should we be using a different concept than "skipped" to describe these?

Reply:
Hm yeah reasonable. Prior art is "unhandled", which leads to:

- `unhandled_subset`
- A asset partition is considered unhandled when it has a reason to be materialized by AMP logic, but has not been requested yet
- An asset partition is considered handled once it is requested by AMP logic, or materialized externally

Still pretty clunky, but maybe a bit better.

This also brings into focus the fact that we also separately keep track of "handled root asset partitions" for the purposes of the materialize_on_missing rule, which feels pretty rough.

This one's tricky because we don't necessarily have a great way of noticing when new partitions pop into being.